tooling/ssa_fuzzer/fuzzer/src/base_context.rs

```diff
@@ -0,0 +1,374 @@
+        let mut acir_ids = vec![];
+        let mut brillig_ids = vec![];
+        for value in values {
+            let field_element = value.into();
+            acir_ids.push(id_to_int(acir_builder.insert_constant(field_element)));
+            brillig_ids.push(id_to_int(brillig_builder.insert_constant(field_element)));
+        }
+        Self {
+            acir_builder,
+            brillig_builder,
+            acir_ids,
+            brillig_ids,
+            acir_arrays: vec![],
+            brillig_arrays: vec![],
+            is_constant: true,
+        }
+    }
+
+    /// Creates a new array from a vector of indices of variables
+    /// Skips non-presented variables
+    pub(crate) fn insert_array(&mut self, elements: Vec<u32>) {
+        let mut acir_values_ids = vec![];
+        let mut brillig_values_ids = vec![];
+        for elem in elements {
+            let acir_len = self.acir_ids.len();
+            let brillig_len = self.brillig_ids.len();
+            acir_values_ids.push(elem % acir_len as u32);
+            brillig_values_ids.push(elem % brillig_len as u32);
+        }
+        let acir_len = acir_values_ids.len();
+        let brillig_len = brillig_values_ids.len();
+        let acir_array = self.acir_builder.insert_make_array(acir_values_ids);
+        let brillig_array = self.brillig_builder.insert_make_array(brillig_values_ids);
+        self.acir_arrays.push(Array { id: acir_array, length: acir_len as u32 });
+        self.brillig_arrays.push(Array { id: brillig_array, length: brillig_len as u32 });
+    }
+
+    /// Gets an element from an array at the given index
+    pub(crate) fn insert_array_get(&mut self, array_idx: u32, index: u32) {
+        if self.acir_arrays.is_empty() {
+            // no arrays created
+            return;
+        }
+        // choose array by index
+        let acir_arrays_len = self.acir_arrays.len() as u32;
+        let brillig_arrays_len = self.brillig_arrays.len() as u32;
+        let acir_array = self.acir_arrays[(array_idx % acir_arrays_len) as usize];
+        let brillig_array = self.brillig_arrays[(array_idx % brillig_arrays_len) as usize];
+        let acir_array_id = acir_array.id;
+        let brillig_array_id = brillig_array.id;
+
+        let acir_id = self.acir_ids[(index % acir_array.length) as usize];
+        let brillig_id = self.brillig_ids[(index % brillig_array.length) as usize];
+        let acir_return_id = self.acir_builder.insert_array_get(acir_array_id, acir_id);
+        let brillig_return_id = self.brillig_builder.insert_array_get(brillig_array_id, brillig_id);
+        self.acir_ids.push(id_to_int(acir_return_id));
+        self.brillig_ids.push(id_to_int(brillig_return_id));
+    }
+
+    /// Sets an element in an array at the given index
+    pub(crate) fn insert_array_set(&mut self, array_idx: u32, index: u32, value: u32) {
+        if self.acir_arrays.is_empty() {
+            // no arrays created
+            return;
+        }
```

Comment:
Perhaps this ties into my earlier question: how can you be sure that `self.brillig_arrays` is also empty?